yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,267 @@
+    player = Player(mark)
+    player.wins = wins
+    return player
+
+
+class LogicSingleActionTest(unittest.TestCase):
+    def test_advance__none_action(self):
+        state = _new_state(board=Board([
+                [None, None, None], [None, None, Mark.X], [None, None, None]]))
+        Logic([ListActionQueue([None]), ListActionQueue([None])]).advance(state)
+        expected_board_cells = [
+                [None, None, None], [None, None, Mark.X], [None, None, None]]
+        self.assertEqual(state.board.cells, expected_board_cells)
+        self.assertEqual(state.step, 0)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, Phase.INROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance__occupy_valid_action(self):
+        state = _new_state()
+        Logic([
+                ListActionQueue([Action.new_occupy(Cell(1, 2))]),
+                ListActionQueue([None])]).advance(state)
+        expected_board_cell = [
+                [None, None, None], [None, None, Mark.X], [None, None, None]]
+        self.assertEqual(state.board.cells, expected_board_cell)
+        self.assertEqual(state.step, 1)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, Phase.INROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance__occupy_invalid_action(self):
+        state = _new_state(board=Board([
+                [None, None, None], [None, None, Mark.X], [None, None, None]]))
+        Logic([
+                ListActionQueue([Action.new_occupy(Cell(1, 2))]),
+                ListActionQueue([None])]).advance(state)
+        expected_board_cell = [
+                [None, None, None], [None, None, Mark.X], [None, None, None]]
+        self.assertEqual(state.board.cells, expected_board_cell)
+        self.assertEqual(state.step, 1)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, Phase.INROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance__surrender_action(self):
+        state = _new_state()
+        Logic([
+                ListActionQueue([Action.new_surrender()]),
+                ListActionQueue([None])]).advance(state)
+        expected_board_cells = [
+                [None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(state.board.cells, expected_board_cells)
+        self.assertEqual(state.step, 0)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, Phase.OUTROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 1)
+
+    def test_win_condition(self):
+        args_and_expect_list = [
+                ([[Mark.X, Mark.X, Mark.X], [None, None, None], [None, None, None]],
+                 Cell(0, 0), True),
```

Comment:
I still don't understand the indentation here. Why is `Cell` indented by 9 spaces relative to the start of `args_and_expect_list`?

Reply:
It is behavior of the IDE. When I try to break line it applies 9 space in lines under the lines with `(` instead of 8 spaces.